src/main/java/org/usfirst/frc1745/deepspace2019/subsystems/vision/Limelight.java

```diff
@@ -1,87 +1,67 @@
-/*----------------------------------------------------------------------------*/
-/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
-/* Open Source Software - may be modified and shared by FRC teams. The code   */
-/* must be accompanied by the FIRST BSD license file in the root directory of */
-/* the project.                                                               */
-/*----------------------------------------------------------------------------*/
 
 package org.usfirst.frc1745.deepspace2019.subsystems.vision;
 
-import edu.wpi.first.networktables.NetworkTable;
+import org.usfirst.frc1745.deepspace2019.NetworkOperations;
+import org.usfirst.frc1745.deepspace2019.subsystems.drive.DrivingDeltas;
 
-/**
- * Add your docs here.
- */
 public class Limelight {
+  /*
+   * Horizontal Offset From Crosshair To Target (-27 degrees to 27 degrees)
+   */
+  private double horizontalTargetOffset = NetworkOperations.getNetworkTable("limelight").getEntry("tx").getDouble(0.0);
+  /*
+   * Vertical Offset From Crosshair To Target (-20.5 degrees to 20.5 degrees)
+   */
+  private double verticalTargetOffset = NetworkOperations.getNetworkTable("limelight").getEntry("ty").getDouble(0.0);
+  /*
+   * Target Area (0% of image to 100% of image)
+   */
+  private double targetArea = NetworkOperations.getNetworkTable("limelight").getEntry("ty").getDouble(0.0);
+  /*
+   * Whether the limelight has any valid targets (0 or 1)
+   */
+  private boolean hasValidTargets = NetworkOperations.getNetworkTable("limelight").getEntry("ty").getDouble(0.0) < 1;
+  
   
   // Constants to be adjusted for calculations
-  private double kpAim = -0.015;
-  private double kpDistance = -0.045;
-  private double minAimCommand = 0;
-  private double deadband = 1.5;
+  private double kpAim = 0.015;
+  private double kpDistance = 0.045;
+  private final double DEADBAND_DEGREES = 2;
 
   // Uses NetworkTable values to calculate speed
-  public double[] calcSpeed(NetworkTable table) {
-
-    double tx = table.getEntry("tx").getDouble(0.0);
-    double ty = table.getEntry("ty").getDouble(0.0);
-
-    double headingError = -tx;
-    double distanceError = -ty;
-    double steeringAdjust = 0.0f;
-
-    if (tx > deadband) {
-      steeringAdjust = (kpAim * headingError) - minAimCommand;
-    } else if (tx < -deadband) {
-      steeringAdjust = (kpAim * headingError) + minAimCommand;
+  public DrivingDeltas calculateDeltas() {
+    double steeringAdjust = 0;
+    double distanceAdjust = 0;
+
+    if(hasValidTargets) {
+      if(Math.abs(horizontalTargetOffset) > DEADBAND_DEGREES) {
+        steeringAdjust = kpAim * horizontalTargetOffset;
+      }
+
+      if(Math.abs(verticalTargetOffset) > DEADBAND_DEGREES) {
+        distanceAdjust = kpDistance * verticalTargetOffset;
+      }
     }
```

Comment:
Here I took out the minAimCommand stuff and added a deadband. I want to try out doing the distanceAdjust based on target blob size instead of vertical offset to the crosshair. Whatever works best